.github/workflows/ci-image-build.yml

```diff
@@ -195,19 +194,19 @@ ${{ inputs.do-build == 'true' && inputs.image-tag || '' }}"
         run: >
           breeze ci-image build --tag-as-latest --image-tag "${{ inputs.image-tag }}"
           --python "${{ matrix.python-version }}"
-          --platform "linux/${{ inputs.platform }}"
+          --platform "${{ inputs.platform }}"
         env:
           DOCKER_CACHE: ${{ inputs.docker-cache }}
           INSTALL_MYSQL_CLIENT_TYPE: ${{ inputs.install-mysql-client-type }}
           UPGRADE_TO_NEWER_DEPENDENCIES: ${{ inputs.upgrade-to-newer-dependencies }}
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          BUILDER: ${{ inputs.platform == 'amd64' && 'default' || 'airflow_cache' }}
+          BUILDER: "airflow_cache"
```

Comment:
That was the clue. BUILDER should be always set to the same `airflow_cache` and then the cache will be nicely reused and rebuilt.